### PR TITLE
Version 4.0.0

### PR DESCRIPTION
**Pull request by vegaro**
RevenueCat Flutter SDK v4 is here!! 

Dancing cats

[Full Changelog](https://github.com/revenuecat/purchases-flutter/compare/3.10.0...4.0.0)

⚠️ ⚠️ Important ⚠️ ⚠️ Observer mode for Amazon is not supported yet.

#### Amazon Appstore Support
We have introduced support for using the Amazon Appstore. We have extensively tested this, and there are some apps using our pre-release Amazon versions in production.

However, we have found some inconsistencies in the way Amazon Appstore prices are reported. We are actively working on patching these inconsistencies.

Please help us help you by reporting any issues you find. [New RevenueCat Issue](https://github.com/RevenueCat/purchases-flutter/issues/new/).

You can enable Amazon Appstore support by configuring the SDK using the new `RevenueCatAPIKeyAmazon` field.

For more information around configuration please take a look at the [Amazon Appstore section in our docs](https://docs.revenuecat.com/docs/amazon-platform-resources). The official [Amazon In-App Purchasing docs](https://developer.amazon.com/docs/in-app-purchasing/iap-overview.html) also contain very valuable information, especially around testing and best practices.

#### StoreKit 2 support
This version of the SDK automatically uses StoreKit 2 APIs under the hood only for APIs that the RevenueCat team has determined work better than StoreKit 1. 

#### New types and cleaned up naming
New types that wrap native types from Apple, Google and Amazon, and we cleaned up the naming of other types and methods for a more consistent experience. 

### Deprecated APIs
| 3.x | 4.0.0 |
| :-: | :-: |
| `Purchases.setup` | `Purchases.configure` |

### Removed APIs
| 3.x | 4.0.0 |
| :-: | :-: |
| `IntroductoryPrice.periodUnit` (`String`) | `IntroductoryPrice.periodUnit` (`PeriodUnit`) |
| `IntroductoryPrice.introPrice` | `IntroductoryPrice.price` |
| `IntroductoryPrice.introPriceString` | `IntroductoryPrice.priceString` |
| `IntroductoryPrice.introPricePeriod` | `IntroductoryPrice.period` |
| `IntroductoryPrice.introPricePeriodNumberOfUnits` | `IntroductoryPrice.periodNumberOfUnits` |
| `IntroductoryPrice.introPriceCycles` | `IntroductoryPrice.cycles` |
| `Purchases.addAttributionData` | `Set<NetworkID>` |
| `Purchases.identify` | `Purchases.logIn` |
| `Purchases.createAlias` | `Purchases.logIn` |
| `Purchases.reset` | `Purchases.logOut` |
| `PurchasesAttributionNetwork` | `Set<NetworkID>` |

### Renamed APIs
| 3.x | 4.0.0 |
| :-: | :-: |
| `PurchaserInfo` | `CustomerInfo` |
| `Transaction` | `StoreTransaction` |
| `Product` | `StoreProduct` |
| `Discount` | `StoreProductDiscount` |
| `PaymentDiscount` | `PromotionalOffer` |
| `Package.product` | `Package.storeProduct` |
| `Purchases.PurchaserInfoUpdateListener` | `Purchases.CustomerInfoUpdateListener` |
| `Purchases.removePurchaserInfoUpdateListener` | `Purchases.removeCustomerInfoUpdateListener` |
| `Purchases.restoreTransactions` | `Purchases.restorePurchases` |
| `Purchases.getPurchaseInfo` | `Purchases.getCustomerInfo` |
| `Purchases.invalidatePurchaserInfoCache` | `Purchases.invalidateCustomerInfoCache` |
| `Purchases.getPaymentDiscount` | `Purchases.getPromotionalOffer` |

### Moved classes
- `Store` enum has been moved from `entitlement_info_wrapper.dart` to `store.dart`

### Other
* Support fastlane plugin revenuecat internal (#410) via Toni Rico (@tonidero)
* Update pubspec.yaml (#406) via Corey Rabazinski (@CoreyRab)
* Update bug_report.md (#402) via Joshua Liebowitz (@taquitos)

### CODE REVIEW

File: CHANGELOG-LATEST.md [hunk 1]
```diff
@@ -62,3 +63,8 @@ New types that wrap native types from Apple, Google and Amazon, and we cleaned u
 
 ### Moved classes
 - `Store` enum has been moved from `entitlement_info_wrapper.dart` to `store.dart`
+
+### Other
+* Support fastlane plugin revenuecat internal (#410) via Toni Rico (@tonidero)
+* Update pubspec.yaml (#406) via Corey Rabazinski (@CoreyRab)
+* Update bug_report.md (#402) via Joshua Liebowitz (@taquitos)
```
Comment (tonidero) [inline CHANGELOG-LATEST.md:70]: Hmm I wonder if we should add these to the changelog, don't think they add much value... But well, it's NABD

File: CHANGELOG.md [hunk 2]
```diff
@@ -237,7 +310,7 @@ The purchase modal will not show correctly for Amazon in these environments.
 ## 3.4.5
 
 - Bumped purchases-android to 4.3.1 [Changelog here](https://github.com/RevenueCat/purchases-android/releases/4.3.1),
-which fixes canMakePayments not returning (see related issue: https://github.com/RevenueCat/purchases-unity/issues/61)
+which fixes canMakePayments not returning (see related issue: https://github.com/RevenueCat/purchases-flutter/issues/61)
```
Comment (tonidero) [inline CHANGELOG.md:313]: Hmm just want to confirm this is intentional?

**Review by tonidero — APPROVED** [review]
LGTM! Just some questions.

File: CHANGELOG-LATEST.md (hunk 1 above)
Comment (vegaro) [reply CHANGELOG-LATEST.md:70]: I wasn't sure either so if you thought the same, I will remove them haha

File: CHANGELOG.md (hunk 2 above)
Comment (vegaro) [reply CHANGELOG.md:313]: Reverting. Actually neither https://github.com/RevenueCat/purchases-unity/issues/61 nor https://github.com/RevenueCat/purchases-flutter/issues/61 seem to be the correct one and they look unrelated to  "canMakePayments not returning"

**Pull request merged**